packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -89,91 +301,48 @@ def create_client(
     built_in_import_modules: List[str] = []
     custom_import_modules: List[str] = []
 
-    if all:
-        measurement_service_class = get_all_registered_measurement_service_classes(discovery_client)
-        if len(measurement_service_class) == 0:
-            raise click.ClickException("No registered measurements.")
-    else:
-        if not measurement_service_class:
-            raise click.ClickException(
-                "The measurement service class cannot be empty. Either provide a measurement service class or use the 'all' flag to generate clients for all registered measurements."
-            )
-
-    if directory_out is None:
-        directory_out_path = pathlib.Path.cwd()
+    has_batch_parameters = (
+        measurement_service_class or all or module_name or class_name or directory_out
+    )
```

Comment:
https://pypi.org/project/click-option-group/ could be helpful for this.
- It allows displaying options under a subheading such as "Batch mode configuration:"
- It can enforce that a group of options are mutually exclusive (such as `--interactive`, `--service-class`, and `--all`).
- It can enforce that at least one option from the group is set.
- I'm not sure, but it may be possible to make a group conflict with another group. @mshafer-NI said he'll look this up on Monday.

Reply:
We are working on the `optgroup` approach.